src/tools/phase_angle.py

```diff
@@ -1,35 +1,54 @@
-from numpy import angle, around, mod, pi, log2
+from numpy import angle, around, mod, pi, log2, ndarray
+from numpy.typing import NDArray
 from .base import convert_state
+from typing import Union
+from ..quantum_circuit import QuantumCircuit
+
+
+def phase_angle(
+    quantumstate: Union[ndarray, QuantumCircuit],
+    show_bit: Union[str, int] = -1,
+    round: int = 3,
+    radian: bool = True,
+) -> NDArray:
+    """Outputs the phase angle of a quantum circuit state.
+
+    ```
+    from qcpy import quantumcircuit, phaseangle
+    phaseangle(quantumcircuit(qubits = 2)
+    ```
+    Args:
+        state: (ndarray/quantumcircuit, required): State to convert to a phase angle array. No default.
+        show_bit: (str/int, optional): Can specify showing a single value from the state. Defaults to -1.
+        round: (int, optional): Rounds the phase angle array by a given value. Defaults to 3.
+        radian: (bool, true): Can represent the phase angle array in radians or not. Defaults to true.
+    Returns:
+        NDArray: Phase angle array from given state.
+    """
+    state = convert_state(quantumstate)
+    size = int(log2(quantumstate.size))
 
-
-def phase_angle(state, show_bit=-1, round: int = 3, radian: bool = True):
-
-    size = int(log2(state.size))
-
-    state = convert_state(state)
     if round < 0:
-        exit(
-            f"Error: QuantumCircuit.tools.phaseAngle() -- round placement must be a value greater than 0."
-        )
+        RoundBelowZero(f"Cannot round to {round} needs to be 0 or greater")
```

Comment:
correct me if im wrong but wouldn't you want to do `raise RoundBelowZero()`? Maybe I didn't know you can raise an error in python without explicitly using the keyword `raise`